### PR TITLE
test(internal): check that ddtrace does not force-load lazy modules

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

This change adds a test to ensure that the import of ddtrace does not cause lazy modules to be force-loaded when the ModuleWatchdog is installed on start-up.

## Checklist
- [x] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: tests/internal/test_module.py [hunk 1]
```diff
@@ -339,3 +339,39 @@ def test_module_watchdog_dict_shallow_copy():
     assert original_modules == new_modules
 
     ModuleWatchdog.uninstall()
+
+
+@pytest.mark.skipif(sys.version_info < (3, 5), reason="requires python3.5 or higher")
```
Comment (mabdinur) [inline tests/internal/test_module.py:344]: Can we expand on this reason? Why is python2.7 not supported

File: tests/internal/test_module.py [hunk 2]
```diff
@@ -339,3 +339,39 @@ def test_module_watchdog_dict_shallow_copy():
...
+@pytest.mark.subprocess(out="ddtrace imported\naccessing lazy module\nlazy loaded\n")
+def test_module_watchdog_no_lazy_force_load():
```
Comment (mabdinur) [inline tests/internal/test_module.py:346]: Should this test contain assertions for when `ddtrace imported`, `lazy loaded`, and `accessing lazy module` are logged? It's a little unclear how this test could fail

File: tests/conftest.py [hunk 3]
```diff
@@ -231,6 +232,13 @@ def _subprocess_wrapper():
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_runtest_protocol(item):
+    if item.get_closest_marker("skip"):
```
Comment (mabdinur) [inline tests/conftest.py:235]: Can we add a comment on why we check the `skip` marker first

**Review by mabdinur — COMMENTED** [review]
left a few questions and nits. Feel free to resolve

File: tests/internal/test_module.py (hunk 1 above)
Comment (P403n1x87) [reply tests/internal/test_module.py:344]: [LazyLoader](https://docs.python.org/3/library/importlib.html#importlib.util.LazyLoader) was introduced in 3.5. I'll change the reason to mention this.

File: tests/conftest.py (hunk 3 above)
Comment (P403n1x87) [reply tests/conftest.py:235]: oh there's no particular reason apart from checking the simplest case first.

**Review by mabdinur — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**